File: string/sort-the-people/sort-the-people.ts
function sortPeople(names: string[], heights: number[]): string[] {
    const lookup : {[key: number]: string} = {};

    for (let i = 0; i < names.length; i++) {
        lookup[heights[i]] = names[i];
    }

    heights.sort((a, b) => b - a);

    for (let i = 0; i < names.length; i++) {
        names[i] = lookup[heights[i]];
    }

    return names;
};

File: string/sort-the-people/sort-the-people.go
package sortthepeople

import "sort"

func sortPeople(names []string, heights []int) []string {
	lookup := make(map[int]string)

	for i := 0; i < len(names); i++ {
		lookup[heights[i]] = names[i]
	}

	sort.Ints(heights)

	j := len(names) - 1
	for i := 0; i < len(names); i++ {
		names[i] = lookup[heights[j-i]]
	}

	return names
}
